lib/srv/reexec.go

```diff
@@ -94,6 +119,11 @@ type ExecCommand struct {
 
 	// X11Config contains an xauth entry to be added to the command user's xauthority.
 	X11Config X11Config `json:"x11_config"`
+
+	// ExtraFilesLen is the number of extra files that are inherited from
+	// the parent process. These files start at file descriptor 3 of the
+	// child process, and are only valid for processes without a terminal.
+	ExtraFilesLen int `json:"extra_files_len"`
```

Comment:
I guess the open file handles in `/proc/self/fd` could be used instead of this field, but I can foresee potential issues where somehow a file handle is create by the process running `RunCommand` before `/proc` is queried, but is closed before the child process created causing an error, or extra files getting inherited. 

Explicitly passing how many files will be inherited seems more reliable to me.